src/main/java/frc/robot/commands/ChargeCommand.java

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.SwerveSubsystem;
+
+public class ChargeCommand extends CommandBase {
+  ProfiledPIDController pidControlX;
+  double currentX;
+  double desiredX;
+  double deltaX;
+
+  double rollValue;
+  double initRollValue;
+
+  double ROLL_TOLERANCE = 15;
+  /** Creates a new ChargeCommand. */
+  public ChargeCommand( double deltaX) {
+    pidControlX = new ProfiledPIDController(1, 0.0, 0.2, 
+    new TrapezoidProfile.Constraints(1, 1));
+
+    this.deltaX = deltaX;
+    // Use addRequirements() here to declare subsystem dependencies.
+    addRequirements(SwerveSubsystem.getInstance());
+    
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    currentX = SwerveSubsystem.getInstance().getPose().getX();
+    desiredX = currentX + deltaX;
+
+    //reset current positions
+    pidControlX.reset(currentX);
+
+    //get the initial roll value
+    initRollValue = SwerveSubsystem.getInstance().getRollValue();
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
```

Comment:
We might want to add heading control to make sure the heading doesn't slip/slide when it goes onto the plastic.

Aka, copy paste the heading control the TranslationCommand as well. The Y is less important and can be left out.